libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/route/RouteEngine.java

```diff
@@ -35,15 +36,15 @@ public void fetchRoute(Point origin, RouteProgress routeProgress) {
     this.routeProgress = routeProgress;
 
     // Calculate remaining waypoints
-    List<Point> coordinates = new ArrayList<>(routeProgress.directionsRoute().routeOptions().coordinates());
```

Comment:
This method seems really similar to the one in `RouteViewModel`, maybe some of it could be extracted into `NavigationRoute`? Or maybe somewhere else

Reply:
Yeah definitely a lot of duplicate here, good call